java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java

```diff
@@ -104,17 +104,10 @@
    */
   static int findColumns(SchemaEvolution evolution,
```

Comment:
For this one, we hit many issues at Apache ORC 1.6.x due to the breaking API and behavior change.
Although this is an implementation internal class, Hive and Spark's hive module assumes old behavior.
So, if we need this change, I'd like to recommend to have a configuration to support both behavior.

Reply:
The behavior change we did to this API is as follows:
* If the column is present is missing in readSchema then throw an exception of missing column
* If the column is missing in file but present in readSchema then we see this as missing column in file handled via NullReader

Wanted to confirm that the ask is to not throw an exception when Hive/Spark use a column name that is missing in the readSchema. I might be missing the scenario where this is the case, I was under the impression that column references should always be valid within the readSchema.

Reply:
I believe it is safe to assume that the column should always be part of the readerSchema -- if thats not the case and there is a column/schema missmatch the caller (Spark/Hive) will be responsible for handling the RuntimeException.

I like this approach because its cleaner (and distinguishes between the two cases) so +1 for that -- but lets first make sure we add the behavior change as part of the method doc.

Reply:
@dongjoon-hyun My guess is that there are some unit tests that need fixing. I had to do the same here in ORC that is the reason for that new `checkEvolutionPosn` method in the test class.

@pgaref Updated the javadoc for the method.